### ISSUE TITLE
Running sample notebook fails with error "Permission Denied"

### ISSUE THREAD

**Issue by nluetts** [brianguenter/OpticSim.jl#41]
**Describe the bug**
Running `OpticSim.NotebooksUtils.run_sample("EmittersIntro.jl")` fails with error "Permission Denied"

**To Reproduce**

```bash
mkdir myproject && cd myproject
julia
```
```julia
julia> ]activate .
julia> ]add OpticSim
julia> using OpticSim
julia> OpticSim.NotebooksUtils.run_sample("EmittersIntro.jl")
┌ Info: Copying the sample notebook [EmittersIntro.jl] from the samples directory to the current one
│ /home/nluetts/.julia/packages/OpticSim/gelEh/samples/notebooks/EmittersIntro.jl ->
└ /tmp/myproject/EmittersIntro.jl
[ Info: Running sample notebook from [/tmp/myproject/EmittersIntro.jl]
[ Info: Launching Notebook [/tmp/myproject/EmittersIntro.jl]
ERROR: SystemError: opening file "/tmp/myproject/EmittersIntro.jl": Permission denied
```

**Expected behavior**
Running `OpticSim.NotebooksUtils.run_sample("EmittersIntro.jl")` opens Pluto and let's you play with the example

**Desktop (please complete the following information):**
 - OS: Fedora 40
 - Julia version 1.10.6
 - OpticSim.jl version 0.6.0

**Additional context**

The problem is that the created file has no write permissions:

```bash
[nluetts@fedora myproject]$ ll
total 292
-rw-r--r--. 1 nluetts nluetts 95212 Nov 15 10:21 'EmittersIntro backup 1.jl'
-r--r--r--. 1 nluetts nluetts 95212 Nov 15 10:21  EmittersIntro.jl
-rw-r--r--. 1 nluetts nluetts 94886 Nov 15 10:13  Manifest.toml
-rw-r--r--. 1 nluetts nluetts    57 Nov 15 10:13  Project.toml
```

Manually setting `chmod +w EmittersIntro.jl` fixes the problem.

I guess you could add a call to [Base.Filesystem.chmod](https://docs.julialang.org/en/v1/base/file/#Base.Filesystem.chmod) after this line:

https://github.com/brianguenter/OpticSim.jl/blob/c15efe4450abd137914ef54df44eb0490d03773c/src/NotebooksUtils/NotebooksUtils.jl#L139

but I am not sure if that would be cross-platform.

**Comment by brianguenter** [comment]
Have you tried that fix? If it works submit a PR and I'll test it on my machine. Then we'll know it works on Fedora and Windows.

**Comment by nluetts** [comment]
Yes, it fixes the error, but the notebook does not run through, it errors out when loading PlutoUI.jl and DataFrames.jl, because those are not part of my test project. So I guess you should opt into the Pluto package management or create a folder including a Project.toml/Manifest.toml + the sample notebook instead of merely copying the notebook file.

**Comment by brianguenter** [comment]
Sorry for the delayed response, we had a bomb cyclone hit our state. Power and internet went out for 500,000 people. 

Last time I checked Pluto package management could be extremely slow, since it didn't cache anything from previous runs of Pluto. Has this changed with the release of 1.11?

**Comment by brianguenter** [comment]
Pluto Notebooks are no longer a part of the core OpticSim build.
